### PR TITLE
Fix mypy tests for `eth` package

### PR DESCRIPTION
**Pull request by hwwhww**
### What was wrong?
Mypy tests were not being triggered for `eth` module.

### How was it fixed?
1. Fixed CircleCI setting
2. Fixed some type hints errors

#### Cute Animal Picture

halloween

### CODE REVIEW

File: eth/vm/forks/byzantium/__init__.py [hunk 1]
```diff
@@ -74,7 +74,8 @@ class ByzantiumVM(SpuriousDragonVM):
     compute_difficulty = staticmethod(compute_byzantium_difficulty)
     configure_header = configure_byzantium_header
     make_receipt = staticmethod(make_byzantium_receipt)
-    get_uncle_reward = staticmethod(get_uncle_reward(EIP649_BLOCK_REWARD))
+    get_uncle_reward = get_uncle_reward(EIP649_BLOCK_REWARD)
+    get_uncle_reward = staticmethod(get_uncle_reward)
```
[the remark this answers is not in the document]
Comment (pipermerriam) [reply eth/vm/forks/byzantium/__init__.py]: by works you mean why `mypy` is ok with it or why it works in the context of python.

File: eth/utils/blake.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from hashlib import blake2b
+from hashlib import blake2b  # type: ignore
```
Comment (pipermerriam) [inline eth/utils/blake.py]: Can you add a comment explaining why we have to `type: ignore` this.

File: eth/vm/forks/byzantium/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (hwwhww) [reply eth/vm/forks/byzantium/__init__.py]: For `mypy`. 
The previous error message was:
```sh
eth/vm/forks/byzantium/__init__.py:77: error: Incompatible types in assignment (expression has type "staticmethod", base class "BaseVM" defined the type as "Callable[[int, BaseBlock], int]")
eth/vm/forks/constantinople/__init__.py:34: error: Incompatible types in assignment (expression has type "staticmethod", base class "BaseVM" defined the type as "Callable[[int, BaseBlock], int]")
```

[the remark this answers is not in the document]
Comment (pipermerriam) [reply eth/vm/forks/byzantium/__init__.py]: Yeah, that's weird.  This deserves a comment explaining why, and maybe a minute checking the `mypy` github issues to see if this is a known bug/deficiency.

[the remark this answers is not in the document]
Comment (hwwhww) [reply eth/vm/forks/byzantium/__init__.py]: Looks like a bug of  `mypy`: https://github.com/python/mypy/issues/5530
@pipermerriam  do you think it's better to keep this current fix + comment the issue,  or add `# type: ignore` + comment?

[the remark this answers is not in the document]
Comment (pipermerriam) [reply eth/vm/forks/byzantium/__init__.py]: fix + comment seems best.

**Pull request merged**